File: internal/models/domain.go
package models

import "time"

type DomainStatus string

const (
	DomainStatusPending  DomainStatus = "pending"
	DomainStatusVerified DomainStatus = "verified"
	DomainStatusActive   DomainStatus = "active"
	DomainStatusFailed   DomainStatus = "failed"
	DomainStatusRevoked  DomainStatus = "revoked"
)

type Domain struct {
	Base
	OrganizationID       string       `json:"organizationId" gorm:"type:uuid;not null;index"`
	TunnelID             *string      `json:"tunnelId,omitempty" gorm:"type:uuid;index"`
	Hostname             string       `json:"hostname" gorm:"uniqueIndex;not null"`
	Status               DomainStatus `json:"status" gorm:"type:varchar(20);not null;default:'pending';index"`
	VerificationMethod   string       `json:"verificationMethod" gorm:"type:varchar(20);not null"`
	VerificationToken    string       `json:"-" gorm:"not null"`
	VerifiedAt           *time.Time   `json:"verifiedAt,omitempty"`
	CertificateStatus    string       `json:"certificateStatus" gorm:"type:varchar(20);not null;default:'pending'"`
	CertificateExpiresAt *time.Time   `json:"certificateExpiresAt,omitempty"`
}

File: apps/web/src/interfaces/domain.ts
import type { Entity } from '#/interfaces/api';

export type DomainStatus =
  | 'pending'
  | 'verified'
  | 'active'
  | 'failed'
  | 'revoked';

export type Domain = Entity & {
  organizationId: string;
  tunnelId?: string;
  hostname: string;
  status: DomainStatus;
  verificationMethod: string;
  verifiedAt?: string;
  certificateStatus: string;
  certificateExpiresAt?: string;
};
